packages/apt/build.sh

```diff
@@ -54,37 +50,45 @@ share/man/man1/apt-transport-mirror.1
 share/man/man8/apt-cdrom.8
 "
 
-termux_step_pre_configure() {
-	# Certain packages are not safe to build on device because their
-	# build.sh script deletes specific files in $TERMUX_PREFIX.
-	if $TERMUX_ON_DEVICE_BUILD; then
-		termux_error_exit "Package '$TERMUX_PKG_NAME' is not safe for on-device builds."
-	fi
+TERMUX_PKG_ON_DEVICE_BUILD_NOT_SUPPORTED=true
 
+termux_step_pre_configure() {
 	# Fix i686 builds.
 	CXXFLAGS+=" -Wno-c++11-narrowing"
 	# Fix glob() on Android 7.
 	LDFLAGS+=" -Wl,--no-as-needed -landroid-glob"
 
 	# for manpage build
-	local docbook_xsl_version=$(. $TERMUX_SCRIPTDIR/packages/docbook-xsl/build.sh; echo $TERMUX_PKG_VERSION)
+	local docbook_xsl_version
+	docbook_xsl_version=$(. "$TERMUX_SCRIPTDIR/packages/docbook-xsl/build.sh"; echo "$TERMUX_PKG_VERSION")
 	TERMUX_PKG_EXTRA_CONFIGURE_ARGS+=" -DDOCBOOK_XSL=$TERMUX_PREFIX/share/xml/docbook/xsl-stylesheets-$docbook_xsl_version-nons"
 }
 
 termux_step_post_make_install() {
-	{
-		echo "# The main termux repository, with cloudflare cache"
-		echo "deb https://packages-cf.termux.dev/apt/termux-main/ stable main"
-		echo "# The main termux repository, without cloudflare cache"
-		echo "# deb https://packages.termux.dev/apt/termux-main/ stable main"
-	} > $TERMUX_PREFIX/etc/apt/sources.list
+	mkdir -p "$TERMUX_PREFIX/etc/apt/sources.list.d"
+	{ # write deb822 source entries for the main repo
+	printf '%s\n' \
+		"# The main termux repository, with cloudflare cache" \
+		"Types: deb" \
+		"URIs: https://packages-cf.termux.dev/apt/termux-main/" \
+		"Suites: stable" \
+		"Components: main" \
+		"Signed-By: $TERMUX_PREFIX/etc/apt/trusted.gpg.d/grimler.gpg, $TERMUX_PREFIX/etc/apt/trusted.gpg.d/termux-autobuilds.gpg" \
+		"" \
+		"# The main termux repository, without cloudflare cache" \
+		"#Types: deb" \
+		"#URIs: https://packages-cf.termux.dev/apt/termux-main/" \
+		"#Suites: stable" \
+		"#Components: main" \
+		"#Signed-By: $TERMUX_PREFIX/etc/apt/trusted.gpg.d/grimler.gpg, $TERMUX_PREFIX/etc/apt/trusted.gpg.d/termux-autobuilds.gpg"
+	} > "$TERMUX_PREFIX/etc/apt/sources.list.d/main.sources"
```

Comment:
I think we need to update `pkg` in the same PR as well, and depend on apt >= 3.0.x. 

`pkg` still modifies `$PREFIX/etc/apt/sources.list`, and is unaware of the .list -> .sources rename so the tool does not work as intended any more. I will work on it after work today

Reply:
I'd say we should update it in the same PR AND write `TERMUX_PKG_{DEPENDS,CONFLICTS,BREAKS}` rules to prevent mixing them in any possible situation.

Reply:
Since `pkg` is part of `termux-tools` I don't have it in this PR.
- termux/termux-tools#165 implements Deb822 support in `pkg` (and `termux-info`).
It should still retain compatibility with the legacy `sources.list` format so requiring Apt >= 3.0.0 shouldn't be necessary.

Reply:
- I'm marking this as resolved since this part of the conversation has been superceded by comments in termux/termux-tools#165 and the rest of this PR.